mkosi/__init__.py

```diff
@@ -2168,30 +2165,36 @@ def copy_uki(context: Context) -> None:
     if (uki := context.root / efi_boot_binary(context)).exists() and (
         KernelType.identify(context.config, uki) == KernelType.uki
     ):
-        pass
+        return uki
     elif (uki := context.root / shim_second_stage_binary(context)).exists() and (
         KernelType.identify(context.config, uki) == KernelType.uki
     ):
-        pass
-    elif ukis:
-        uki = ukis[0]
-    else:
-        return
+        return uki
 
-    shutil.copy(uki, context.staging / context.config.output_split_uki)
+    if not ukis:
+        return None
 
-    # Extract the combined initrds from the UKI so we can use it to direct kernel boot with qemu if needed.
-    extract_pe_section(context, uki, ".initrd", context.staging / context.config.output_split_initrd)
+    return ukis[0]
 
-    # ukify will have signed the kernel image as well. Let's make sure we put the signed kernel
-    # image in the output directory instead of the unsigned one by reading it from the UKI.
-    extract_pe_section(context, uki, ".linux", context.staging / context.config.output_split_kernel)
+
+def copy_uki(context: Context) -> None:
+    if (context.staging / context.config.output_split_uki).exists():
+        return
```

Comment:
Can't we just return the path here and then get rid of get_uki() and use copy_uki() everywhere instead? I'd prefer to have this be a single function instead of two.

Reply:
That would result in the uki always getting copied when it exists, even when it wasn't specified. Just conditioning the copy part and still return the path would feel a bit weird, but I could do that.

Reply:
I think it's good as is with the name change. I don't think `copy_uki` shouldn't copy the uki in some scenarios, that would just be false advertising. I could see just always copying out the UKI and having the kernel and initrd be options in addition to the uki. I think that would also be fair, but I lean more towards the way it is now.